### PR TITLE
Random Battles updates

### PR DESCRIPTION
**Pull request by ACakeWearingAHat**
Gen 8:
-Lucario: Stone Edge over Ice Punch, adjust corresponding hardcode to match
-Silvally-Fairy: Flame Charge over Fire Fang
-Zeraora: Blaze Kick over Fire Punch; the power is important for several KOs
-Fix Rillaboom and G-Max Rillaboom's level difference
-Eldegoss: -Charm
-Prevent Shift Gear + U-turn Genesect again
-Create new focus sash qualifications s.t. pokemon with 250 total bulk (ex. Mamoswine) and below will get sash with hazards in lead slot and Pokemon with 210 total bulk (ex. Indeedee-M) and below will get focus sash without hazards in the lead slot.
-Rewrite the Cinccino hardcode so it both is more explicitly about cinccino and prevents focus sash from generating

Gen 7:
-Stone Edge over Fire Punch on hitmonchan for better coverage
-remove Memento on dugtrio to prevent leftovers from generating
-trim Primeape, Palkia, Cobalion, and Tapu Lele's movepools to increase overall set quality
-Pursuit should no longer generate as the only Dark move on dual-typed dark-types
-Nihilego should get SpA beast boosts
-force Eruption on Fire-types that have it (typhlosion)
-Rotom-Fan will now actually always have air slash
-Azumarill will have forced aqua jet now

### CODE REVIEW

File: data/formats-data.ts [hunk 1]
```diff
@@ -5471,7 +5471,7 @@ export const FormatsData: {[k: string]: SpeciesFormatsData} = {
 	},
 	rillaboom: {
 		randomBattleMoves: ["grassyglide", "highhorsepower", "knockoff", "uturn", "woodhammer"],
-		randomBattleLevel: 77,
+		randomBattleLevel: 78,
```
Comment (AnnikaCodes) [inline data/formats-data.ts:5474]: Is this intentional?

File: test/random-battles/gen7.js [hunk 2]
```diff
@@ -13,6 +13,62 @@ describe('[Gen 7] Random Battle', () => {
...
+	});
+
+	it('should give Typhlosion Eruption', () => {
+		testSet('typhlosion', options, set => {
+			assert(set.moves.includes('eruption'), `Typhlosion: got ${set.moves}`);
+		});
+	});
+
+	it('should not generate Pursuit as the only Dark STAB move', () => {
+		testSet('weavile', options, set => {
```
Comment (AnnikaCodes) [inline test/random-battles/gen7.js:29]: GitHub decided to be crappy and lose my half-written comment but this should be programmatically done.

Comment (AnnikaCodes) [reply test/random-battles/gen7.js:29]:
```ts
const dex = Dex.forFormat(options.format);
const darkTypesWithPursuit = dex.species
    .all()
    .filter(pkmn => pkmn.types.includes('Dark') && pkmn.randomBattleMoves.includes('pursuit'))
    .map(pkmn => pkmn.id);
for (const pokemon of darkTypesWithPursuit) {
    testSet(pokemon, options, ...);
}
```

**Comment by ACakeWearingAHat** [comment]
Yeah, that seems good!

**Comment by AnnikaCodes** [comment]
> Yeah, that seems good!

Are you going to implement it yourself, or do you want my help?

File: test/random-battles/gen7.js [hunk 3]
```diff
@@ -13,6 +13,37 @@ describe('[Gen 7] Random Battle', () => {
...
+		testSet('typhlosion', options, set => {
+			assert(set.moves.includes('eruption'), `Typhlosion: got ${set.moves}`);
+		});
+	});
+
+	it('should not generate Pursuit as the only Dark STAB move', () => {
+		const dex = Dex.forFormat(options.format);
+		const darkTypesWithPursuit = dex.species
+			.all()
+			.filter(pkmn => pkmn.types.includes('Dark') && pkmn.randomBattleMoves.includes('pursuit'))
```
Comment (AnnikaCodes) [inline test/random-battles/gen7.js:32]:
```suggestion
			.filter(pkmn => pkmn.types.includes('Dark') && pkmn.randomBattleMoves?.includes('pursuit'))
```

**Comment by AnnikaCodes** [comment]
Thanks!

**Pull request merged**